cpu: Fix looppoint analysis v25

studyztp opened the pull request:
This is a PR for the issue: https://github.com/gem5/gem5/issues/2395

The new probe structure uses ```dynamic_cast``` to check the```ProbeListener```, which caused a casting mismatch with the LoopPoint Analysis ProbeListener, because the LoopPoint Analysis ProbeListener uses ```StaticInstPtr``` instead of ```const StaticInstPtr``` as part of its Arg while the Probe Point that it is listening to, ```Commit```, uses ```const StaticInstPtr```.
This commit fixes the issue.

giactra commented:
Hi, thanks for your contribution. This PR is targeting stable branch but it should actually target the develop branch:

https://github.com/gem5/gem5/blob/stable/CONTRIBUTING.md

studyztp commented:
> Hi, thanks for your contribution. This PR is targeting stable branch but it should actually target the develop branch:
> 
> https://github.com/gem5/gem5/blob/stable/CONTRIBUTING.md

Thank you for pointing this out. I rebased my branch. I planned to have this as a hotfix for the stable branch, so I targeted the stable branch. Sorry about this.

powerjg commented on test-script/test-looppoint-analysis-script.py:
````diff
@@ -0,0 +1,500 @@
+
+```
+scons build/X86/gem5.opt
+./build/X86/gem5.opt \
````
Does it have to be `X86`? Can we start making all examples use `ALL`?

studyztp replied:
I'll use ALL for all the future examples.

powerjg commented on test-script/test-looppoint-analysis-script.py:
```diff
@@ -0,0 +1,500 @@
+# Copyright (c) 2025 The Regents of the University of California
```
For this whole file... Can you ask copilot to put in comments? This file is going to be impossible for our future selves to be able to debug without any comments. 

You may also want to ask copilot to write the tests for you.

Finally, it may be a better idea to separate out the "tests" into their own file instead of mixing simulation details in with the test code.

studyztp replied:
I used copilot to put in comments, but I asked it to make only necessary comments to prevent polluting the file. I'll let it make more detailed comments for the future test. Thanks for the suggestion.
I think I should make a better test set for the LoopPoint analysis module and separate the tests from this commit. I removed this test script and left only the fix to this commit.

BobbyRBruce commented:
Just to clarify here: what's the status of these tests? The PR adds them, then removes them. Is the intent that I re-add them as testlib tests on this PR?

studyztp committed:
cpu: fix LoopPoint Analysis probe Arg issue

The new probe structure uses dynamic_cast to check the probelistener,
and it caused casting unmatch with the LoopPoint Analysis probelistener
because the LoopPoint Analysis probelistener uses StaticInstPtr instead
of const StaticInstPtr as part of its Arg.
This commit fixed this issue.

studyztp commented:
> Just to clarify here: what's the status of these tests? The PR adds them, then removes them. Is the intent that I re-add them as testlib tests on this PR?

I was intended to add them but now I think I should separate them completely from this PR. Now this PR only has the fix of the LoopPoint Analysis probe. Sorry for my mistake.
